pkg/kubectl/cmd/util/factory.go

```diff
@@ -197,32 +199,57 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clients: clients,
 		flags:   flags,
 
-		Object: func() (meta.RESTMapper, runtime.ObjectTyper) {
+		// If discoverDynamicAPIs is true, make API calls to the discovery service to find APIs that
+		// have been dynamically added to the apiserver
+		Object: func(discoverDynamicAPIs bool) (meta.RESTMapper, runtime.ObjectTyper) {
 			cfg, err := clientConfig.ClientConfig()
 			CheckErr(err)
 			cmdApiVersion := unversioned.GroupVersion{}
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
+			CheckErr(err)
+			if discoverDynamicAPIs {
+				client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
+				CheckErr(err)
+				
+				versions, gvks, err := GetThirdPartyGroupVersions(client.Discovery())
+				CheckErr(err)
+				if len(versions) > 0 {
+					preferredExternalVersion := versions[0]
+
+					accessor := meta.NewAccessor()
+					thirdPartyMapper, err := kubectl.NewThirdPartyResourceMapper(versions, gvks)
+					CheckErr(err)
+					groupMeta := apimachinery.GroupMeta{
+						GroupVersion:  preferredExternalVersion,
+						GroupVersions: versions,
```

Comment:
This doesn't look right.  There's no reason to believe that all third party resources are in the same same group, right?  I think you need to separate by group and then register each group separately.